build fails with ../../snek-gram.ll:258:5: error: ‘for_depth’ undeclared

I am trying to build from source on debian 11 bullseye i386 (old laptop) and aarch64 (raspberry Pi4 with 64bit OS) and on both of them it fails like this
```
make[1]: Entering directory '/home/pi/snek/ports/posix'
lola -DSNEK_SLICE -DSNEK_DICT -o snek-gram.h ../../snek-gram.ll
python3 ../../snek-builtin.py ../../snek-keyword.builtin ../../snek-base.builtin snek-posix.builtin ../../snek-math.builtin ../../snek-input.builtin -o snek-builtin.h
  CC  snek-builtin.o
  CC  snek-code.o
  CC  snek-exec.o
  CC  snek-error.o
  CC  snek-frame.o
  CC  snek-func.o
  CC  snek-lex.o
  CC  snek-list.o
  CC  snek-memory.o
  CC  snek-name.o
  CC  snek-parse.o
In file included from ../../snek-parse.c:219:
../../snek-gram.ll: In function ‘parse’:
../../snek-gram.ll:258:5: error: ‘for_depth’ undeclared (first use in this function)
  258 |     for_depth--;
      |     ^~~~~~~~~
../../snek-gram.ll:258:5: note: each undeclared identifier is reported only once for each function it appears in
In file included from ../../snek-parse.c:219:
../../snek-gram.ll:17:11: error: ‘snek_print_val’ undeclared (first use in this function); did you mean ‘snek_print’?
   17 | command  : @{ snek_print_val = snek_interactive; }@ stat
      |           ^~~~~~~~~~~~~~
      |           snek_print
make[1]: *** [../../snek.defs:163: snek-parse.o] Error 1
make[1]: Leaving directory '/home/pi/snek/ports/posix'
make: *** [Makefile:30: ports/posix/snek] Error 2
```

I can see that mising stuff is declared in C macro PARSE_TOP here https://github.com/keith-packard/snek/blob/main/snek-parse.c#L39 but I don't see the macro being used anywhere so I'm not sure how this is supposed to work. also in generated `ports/posix/snek-gram.h` macro `PARSE_TOP` is not used anywhere.

I installed dependencies including lola as per readme, lola was downloaded as
```
http://deb.debian.org/debian bullseye/main arm64 lola all 1.4-1 [12.0 kB]   
```
I am trying default main branch if that matters.

--- a/README.md
+++ b/README.md
@@ -165,9 +165,19 @@ To build Snek you need these dependencies:
 
 On Debian unstable, you can get everything from the main archive:
 
-	# apt install lola gcc-avr avr-libc python3-serial \
+	# apt install lola gcc-avr avr-libc python3-serial gawk \
           gcc-arm-none-eabi gcc-riscv64-unknown-elf libreadline-dev \
 	  picolibc-arm-none-eabi picolibc-riscv64-unknown-elf
+
+Debian stable has too old a version of lola, so if you're using that,
+you'll need to install lola from the git repository above.
+
+The snek build scripts use the GNU awk implementation, `gawk`,
+including a GNU extension, `strtonum`, which is not available in other
+awk implementations. It's not critical to compiling the code as it's
+only used to print out ROM usage for AVR targets so you can tell how
+much space is still available, but if you don't have gawk, you will
+get build failures.
 
 ### Building and install
 

--- a/snek-builtin.py
+++ b/snek-builtin.py
@@ -96,7 +96,7 @@ class SnekBuiltin:
             if self.value[0].isdigit():
                 return "(snek_poly_t)(float)%s" % self.value
             return self.value
-        name = self.name;
+        name = self.name
         if self.init is not None:
             name = self.init
         return "snek_builtin_%s" % (name.replace(".", "_"))

